subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/exec/ForwardClientInput.java

```diff
@@ -84,4 +84,37 @@ public Void call() {
             throw UncheckedException.throwAsUncheckedException(e);
         }
     }
-}
\ No newline at end of file
+
+    private static class CloseAwareInputStream extends PipedInputStream {
```

Comment:
⭕️ "close aware" doesn't really say what this does on close. Maybe call it something like "EndOnCloseInputStream"

Reply:
Done.